msdm/core/distributions/distributions.py

```diff
@@ -102,30 +106,76 @@ def isclose(
         return True
 
     def marginalize(self, projection: Callable[[Event], Event]):
+        """
+        Marginalize the distribution according to the
+        projection function. `projection` must return hashable values.
+        """
         newdist = defaultdict(lambda : 0)
         for e, p in self.items():
             newdist[projection(e)] += p
         return DictDistribution(newdist)
 
     def expectation(self, real_function: Callable[[Event], float] = lambda e: e):
+        """
+        Return the expected value of real_function under
+        the distribution.
+        """
         tot = 0
         for e, p in self.items():
             tot += real_function(e)*p
         return tot
 
-    def condition(self, predicate: Callable[[Event], bool]):
-        dist = {e: p for e, p in self.items() if predicate(e)}
-        norm = sum(dist.values())
-        for e, p in dist.items():
-            dist[e] = p/norm
+    def condition(self, predicate: Callable[[Event], float]):
```

Comment:
Per convo: should either make this return type a union of float and bool (in python 3.9 I think you can do `float|bool`)

Reply:
Updated 2f39620b98109dda028f3bda3cbd55cde7963022 

I used `Union[float, bool]` since we still test on python 3.7